datatypes/src/primitives/coordinate.rs

```diff
@@ -110,6 +112,27 @@ impl<'c> Into<&'c [f64]> for &'c Coordinate2D {
     }
 }
 
+// TODO: use FixedSizeListBuilder of Float64Builder -> cf. https://issues.apache.org/jira/browse/ARROW-8508
+impl ArrowTyped for Coordinate2D {
+    type ArrowArray = arrow::array::FixedSizeBinaryArray;
+    type ArrowBuilder = arrow::array::FixedSizeBinaryBuilder;
+
+    fn arrow_data_type() -> DataType {
+        arrow::datatypes::DataType::FixedSizeBinary(std::mem::size_of::<Self>() as i32)
+    }
+
+    fn arrow_builder(capacity: usize) -> Self::ArrowBuilder {
+        arrow::array::FixedSizeBinaryBuilder::new(capacity, std::mem::size_of::<Self>() as i32)
+    }
+}
```

Comment:
I am not sure if we did it this way before in our code.

For instance, `BaseRaster` is defined in `datatypes/src/raster/base_raster.rs` and implements `SpatialBounded` in this file.
`SpatialBounded` on the other hand is defined in `datatypes/src/primitives/spatio_temporal_bounded.rs`. But we did not define the implementations in there.

So, what is the guideline to have some traits defined and implemented in one file and the other one separated? I, e.g., implemented the traits FeatureCollection and GeoFeatureCollection in each of the collection types, but not in the files where the Traits are defined. I guess this way you don't have to import the trait file to use the trait (?). But we only have to find a common way to do it.

Reply:
I was thinking of the usage of external types. If we keep the implementation based on external types in a separate file, we can switch and or upgrade external libraries without changing 100 files. However, we still have to find a way to keep the traits clean of external dependencies. While we can not avoid it everywhere, would still prefer this approach.

Reply:
I can't keep the ArrowTyped trait free of external dependencies since I created it to encapsule the external type.
But where to put the file? There is no ArrowType primitive? And to put it into util/arrow also does not make sense.
Furthermore, do I have to put the `impl<'g> Into<geojson::Geometry> for XYZ` also all into one joint file?
Because it is the exact same thing. I guess we have to ask Michael for his opinion to untie the knot and then we proceed this way.